tlatools/org.lamport.tlatools/src/tlc2/tool/impl/SpecProcessor.java

```diff
@@ -580,8 +580,10 @@ private final void processSpec(final Mode mode)
         // the default is to load the first class on the classpath with name tlc2.overrides.TLCOverrides
         // that implements tlc2.overrides.ITLCOverrides.  This is usually the tlc2.overrides.TLCOverrides
         // provided by the CommunityModules.
+        // We also are appending a standard module overrides, so we are able to bring some modules
+        // from CommunityModules without any change, is this a good idea?
         boolean hasCallableValue = false;
-		final String tlcOverrides = System.getProperty("tlc2.overrides.TLCOverrides", "tlc2.overrides.TLCOverrides");
+        final String tlcOverrides = System.getProperty("tlc2.overrides.TLCOverrides", "tlc2.overrides.TLCOverrides") + File.pathSeparator + "tlc2.overrides.TLCStandardOverrides";        
```

Comment:
> Have you considered adopting the (older) approach of how TLC's standard modules are loaded?

Yes, I have tried this before, but it would require considerable modifications of the `Json.java` and `TLCExt.java` files, the "new" approach does the overriding of the operators in what appears to be in a more powerful way.

> we should probably solve the problem of how to deal with multiple TLCOverrides.

The multiple TLCOverrides thing s working, you have to pass the name of the classes using `-Dtlc2.overrides.TLCOverrides` (you can see the comment that you have made in the code above).

> Another question is where to put the additional TLA+ modules. So far, they (Toolbox/Randomization) have gone into StandardModules. Perhaps, we should move TLCExt and JSon to StandardModules too or find a new home for all the modules not mentioned in Lamport's Specifying Systems book.

Right now they are in the `StandardModules` folder, maybe we should create a folder named `NonStandardModules`? About the `TLCStandardOverrides` (and if we are using the new way, which enables us to use the files from CM without a change), we definitely could improve the name, but I see that you may not want this new concept in TLC, what do you say? About the override java files, I'm putting them in `tlc2/overrides/TLCStandardOverrides.java`, do you know a better place?

> We also have to figure out how to deprecate Json and TLCExt in CommunityModules and/or define what happens if a user runs TLC with this PR merged and an older CommunityModules. TLCExt and Json in CommunityModules would replace the ones in TLC. However, we cannot simply prevent this because users should be able to replace all modules of TLC with custom ones.

Yeah, I guess we cannot prevent this, if you are using the CM module and you want to debug a change in the CM Json or TLCExt files, then it could not be obvious to the user why his/her change is not working. I guess the deprecation notice is the best way, it should not break anything if you are only using it for TLC operator overriding (instead of depending in the java class for your own weird project). I can add the deprecation notice there if you allow me.

Reply:
And this line would probably be reserved when we merge this PR so we have the SANY overrides before the user overrides.

`final String tlcOverrides = "tlc2.overrides.TLCStandardOverrides"  + File.pathSeparator + System.getProperty("tlc2.overrides.TLCOverrides", "tlc2.overrides.TLCOverrides");`

Reply:
* Please add `TLCExt.tla` and `Json.tla` to `StandardModules`
* Once we have a build of TLC with TLCExt and Json, we will simply remove TLCExt.tla/.java and Json.tla/.java from CommunityModules
* Last thing to iron out is how to load TLCExt.java and Json.java in TLC so that user-provided ones take precedence

Reply:
> We want users to be able to override built-in modules.

Okay, I have switched it and now `TLCStandardOverrides ` is loaded before anything.

> Please add TLCExt.tla and Json.tla to StandardModules

They are already there, Markus o/

> Last thing to iron out is how to load TLCExt.java and Json.java in TLC so that user-provided ones take precedence

I created a `tlc2.overrides.internal` package so we don't have conflicts if the user adds a `Json` or `TLCExt` class using `tlc2.overrides`, I have tested manually and been able to load the override version from CM independently of the classpath order (I have made the Json.java from TLC and the one from CM behave differently to check the results).